heron/spi/src/java/com/twitter/heron/spi/utils/LauncherUtils.java

```diff
@@ -0,0 +1,185 @@
+    } finally {
+      scheduler.close();
+    }
+
+    return ret;
+  }
+
+  /**
+   * Returns a packing plan generated by configured packing class
+   */
+  public static PackingPlan createPackingPlan(final Config config, final Config runtime) {
+    return getInstance()._createPackingPlan(config, runtime);
+  }
+
+  /**
+   * Creates and initializes scheduler instance
+   *
+   * @return initialized scheduler instances
+   */
+  public static IScheduler getSchedulerInstance(
+      Config config,
+      Config runtime,
+      PackingPlan packing) {
+    return getInstance()._getSchedulerInstance(config, runtime, packing);
+  }
+
+  /**
+   * Builds initial runtime config instance using topology information.
+   *
+   * @return initial runtime config instance
+   */
+  public static Config getPrimaryRuntime(TopologyAPI.Topology topology,
+                                         SchedulerStateManagerAdaptor adaptor) {
+    return Config.newBuilder()
+        .put(Keys.topologyId(), topology.getId())
+        .put(Keys.topologyName(), topology.getName())
+        .put(Keys.topologyDefinition(), topology)
+        .put(Keys.schedulerStateManagerAdaptor(), adaptor)
+        .put(Keys.numContainers(), 1 + TopologyUtils.getNumContainers(topology)).build();
+  }
+
+  private static Config putPackingInRuntimeConfig(Config runtime, PackingPlan packing) {
+    Config ytruntime;
+    if (packing == null) {
+      ytruntime = runtime;
+    } else {
+      ytruntime = Config.newBuilder()
+          .putAll(runtime)
+          .put(Keys.instanceDistribution(), packing.getInstanceDistribution())
+          .put(Keys.componentRamMap(), packing.getComponentRamDistribution())
+          .put(Keys.numContainers(), 1 + packing.getContainers().size())
+          .build();
+    }
+    return ytruntime;
+  }
+
+  private static LauncherUtils getInstance() {
+    return instance;
+  }
+
+  /**
+   * For testing only. Could be used to setup a mock
+   */
+  public static void setInstance(LauncherUtils instance) {
+    LauncherUtils.instance = instance;
+  }
+
+  /**
+   * Creates and initializes scheduler instance
+   *
+   * @return initialized scheduler instances
+   */
+  private IScheduler _getSchedulerInstance(
+      Config config,
+      Config runtime,
+      PackingPlan packing) {
+    String schedulerClass = Context.schedulerClass(config);
+    IScheduler scheduler;
+    try {
+      // create an instance of scheduler
+      scheduler = ReflectionUtils.newInstance(schedulerClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      LOG.log(Level.SEVERE, "Failed to instantiate scheduler", e);
+      return null;
+    }
+
+    Config ytruntime = putPackingInRuntimeConfig(runtime, packing);
+
+    scheduler.initialize(config, ytruntime);
+    return scheduler;
+  }
+
+  /**
+   * This method can be mocked for unit testing.
+   */
+  public PackingPlan _createPackingPlan(final Config config, final Config runtime) {
```

Comment:
The util classes provides static methods. However, static methods are hard to test, mock or inject. So the class has these internal methods used by static methods. The user should be using static methods and not the internal ones. I thought `_` could be a good way to easily identify. However I am open to any suggestions. I could create an internal class also. What do you think?